code/edge/src/androidTest/java/com/adobe/marketing/mobile/NetworkResponseHandlerFunctionalTests.java

```diff
@@ -122,7 +122,8 @@ public void testProcessResponseOnError_WhenInvalidJsonError_doesNotHandleError()
 	}
 
 	@Test
-	public void testProcessResponseOnError_WhenGenericJsonError_dispatchesEvent() throws InterruptedException {
+	public void testProcessResponseOnError_WhenGenericJsonError_noWaitingEvents_dispatchesEvent()
```

Comment:
can you add some description on when this use-case will occur?

Reply:
I've added waiting events for this test but keyed under a different request ID so a parent ID is still not added to the response.